### DISCUSSION TITLE
How does --silent actually work?

### DISCUSSION THREAD

**Discussion by Trinovantes** [vitest-dev/vitest#4155] [Q&A] [answered]
I'm using `--silent` and can see that it does suppress this console log

```js
test('foo' () => {
    console.info('Hello World 1')
})
```

However, when I use logging libraries like pino, it still prints its output
```js
import pino from 'pino'

test('foo', () => {
    const logger = pino()
    console.info('Hello World 1') // doesn't print
    logger.info('Hello World 2') // prints
})
```

I think by default pino uses `process.stdout.fd` to write to stdout

**Comment by sheremet-va** [comment, marked as answer]
`--silent` silences all logs from the console (`console` global is overridden to support this), so other ways of logging bypass that.
